nomad/structs/workload_id.go

```diff
@@ -38,7 +39,7 @@ var (
 	// be safe to use in filenames.
 	//
 	// Reuse validNamespaceName to save a bit of memory.
-	validIdentityName = validNamespaceName
+	validIdentityName = regexp.MustCompile("^[a-zA-Z0-9-//]{1,128}$")
```

Comment:
I don't think the second `/` gets us anything here, as `/` doesn't need escaping:

```suggestion
	validIdentityName = regexp.MustCompile("^[a-zA-Z0-9-/]{1,128}$")
```

Reply:
Fixed in https://github.com/hashicorp/nomad/pull/18591/files#diff-d2ec4e0bb147da1d9abe8f14084958a2d4bc16b5cb1cf28f5a07c7f2bb3331e5R44.

For the record, we realized that using slashes in the identity name would be a problem because of how they're written to disk (`<task dir>/<secrets name>`) it would try to create a new directory.